packages/javascript/bh-shared-ui/src/views/TierManagement/Details/DetailsList.tsx

```diff
@@ -0,0 +1,135 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+import { Button } from '@bloodhoundenterprise/doodleui';
+import { FC, useState } from 'react';
+import { UseQueryResult } from 'react-query';
+import { SortableHeader } from '../../../components';
+import { SortOrder } from '../../../types';
+import { cn } from '../../../utils';
+import { SelectedHighlight, itemSkeletons } from './utils';
+
+type DetailsListItem = {
+    name: string;
+    id: number;
+    count: number;
+};
+
+type DetailsListProps = {
+    title: 'Selectors' | 'Tiers' | 'Labels';
+    listQuery: UseQueryResult<DetailsListItem[], unknown>;
+    selected: number | null;
+    onSelect: (id: number) => void;
+};
+
+/**
+ * @description This component is meant to display the lists for either Tiers, Labels, or Selectors but not the Members list since that is a paginated list that loads more data as a user scrolls.
+ * @param {object} props
+ * @param {title} props.title Limited to 'Selectors' | 'Tiers' | 'Labels' as that is what this component is built for
+ * @param {UseQueryResult} props.listQuery The endpoint call result wrapper from react query that allows us to hook into different states that the fetched data could be in
+ * @param {selected} props.selected The id of the particular entity that is selected for the list. It is used for selected item rendering
+ * @param {(id:number) => void} props.onSelect The click handler that should be called when an item from this list is selected. This is primarily being used to set the selected id state in the parent Details component
+ * @returns The component that displays a list of entities for the tier management page
+ */
+export const DetailsList: FC<DetailsListProps> = ({ title, listQuery, selected, onSelect }) => {
+    const [sortOrder, setSortOrder] = useState<SortOrder>();
+
+    return (
+        <div data-testid={`tier-management_details_${title.toLowerCase()}-list`} className='h-full overflow-y-hidden'>
+            {title !== 'Tiers' ? (
+                <SortableHeader
+                    title={title}
+                    onSort={() => {
+                        if (sortOrder === undefined) {
+                            // first click
+                            setSortOrder('desc');
+                        } else if (sortOrder === 'desc') {
+                            // second click
+                            setSortOrder('asc');
+                        } else if (sortOrder === 'asc') {
+                            // third click
+                            setSortOrder(undefined);
+                        }
+                    }}
```

Comment:
I think you mentioned this during the code review -- the first click of the sortable header keeps the sort order as desc, probably because undefined will sort by desc by default (line 104). My suggestion would be to take out undefined as an option all together and default the sortOrder state as 'desc', that way the header always indicates the sort:
```suggestion
                        if (sortOrder === 'desc') {
                            // second click
                            setSortOrder('asc');
                        } else if (sortOrder === 'asc') {
                            // third click
                            setSortOrder('desc');
                        }
                    }}
```

Reply:
Agreed this sounds good

> I think you mentioned this during the code review -- the first click of the sortable header keeps the sort order as desc, probably because undefined will sort by desc by default (line 104). My suggestion would be to take out undefined as an option all together and default the sortOrder state as 'desc', that way the header always indicates the sort:

Reply:
Updated!